module3/projects/rancid_tomatillos/technical_requirements.md

```diff
@@ -0,0 +1,237 @@
+
+**User Storie**  
+- As a user, I can click a movie, and see that movie's details.
+  - At minimum, I should see see a movie's backdrop image, genres, and overview. You may choose to display more information if you'd like.
+  - When a movie's details are displayed, none of the other movies will be visible, but the page's header is still visible.
+  - When a movie's details are displayed, the user should have a way to return to the main view of all movies (i.e. a home button). This button should not be visible if I am on the home page.
+
+<section class="dropdown">
+### 👀 Preview
+
+Screenshot:
+![iteration 3 screenshot](../../../assets/images/projects/rancid-tomatillos/details_hardcoded.png)
+
+In Motion:
+<div style="position: relative; padding-bottom: 56.25%; height: 0;"><iframe src="https://www.loom.com/embed/72d10684d8d84671baecb864f34b5c3f?sid=fe025c68-5c08-40d6-bf03-ebe63f5e921f" frameborder="0" webkitallowfullscreen mozallowfullscreen allowfullscreen style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;"></iframe></div>
+</section>
+</section>
+
+<section class="note dropdown">
+### A Note About Refactoring
+
+We are about to get into some refactoring iterations. These iterations are designed to require heavy refactoring. It is expected that you will have to re-organize the code you've written. Don't be too attached to the code you have right now - it will need to change! For example, if your goal is to shove Router into your existing code while making as few changes as possible, you're doing it wrong.
+
+If it makes you nervous, remember that you should always be working on a branch. All your previous code is safe and sound on the main branch until you've got your new feature working.  
+</section>
+
+<section class="dropdown">
+### Iteration 4: Refactor with Network Requests
+
+For this iteration, we'll finally refactor our application to use actual data from a backend instead of our mock data files. 🥳
+
+**[Here are the API Docs.](./rancid_tomatillos_api_documentation)**
+
+**User Stories**
+- After deleting the entire `data` folder and fetching instead, all prior user stories still work as expected.
+- When you click on a movie, you should see that movie's details (no longer always showing "Spirited Away").
+- When a user votes, a `PATCH` request is sent so the backend data is updated.
+
+<section class="dropdown">
+### 👀 Preview
+
+Screenshot:
+![iteration 4 screenshot](../../../assets/images/projects/rancid-tomatillos/details.png)
+
+In Motion:
+<div style="position: relative; padding-bottom: 56.25%; height: 0;"><iframe src="https://www.loom.com/embed/87f7179a6e8a440b8432e149424e5c8b?sid=f4fa1691-c0ff-4e6d-9796-fafcc456e008" frameborder="0" webkitallowfullscreen mozallowfullscreen allowfullscreen style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;"></iframe></div>
+</section>
+</section>
+
+<section class="dropdown">
+### Iteration 5: Test with Cypress
```

Comment:
Just thinking about this.. I expect some students, now that we're fullstack, getting really high and mighty about TDD cause we drill it in so hard. Have you ever seen students follow TDD with cypress? I was once on a team with someone that did but truly only that one guy. I'm curious how students will respond to this shift in priority. This is not a piece of feedback...just sharing my thoughts. Ultimately this is a useless comment